mx.py

```diff
@@ -1893,11 +1905,84 @@ def buildJavacLike(self, jdk, project, jvmArgs, javacArgs, disableApiRestriction
             abort('Showing task tags is not currently supported for javac')
         javacArgs.append('-encoding')
         javacArgs.append('UTF-8')
-        javac = self.altJavac if self.altJavac else jdk.javac
         jvmArgs += jdk.java_args
+        self.run(jdk, jvmArgs, javacArgs)
+
+    def run(self, jdk, jvmArgs, javacArgs):
+        javac = self.altJavac if self.altJavac else jdk.javac
         cmd = [javac] + ['-J' + arg for arg in jvmArgs] + javacArgs
         run(cmd)
 
+class JavacCompilerDaemon(JavacCompiler):
```

Comment:
Minor: Got confused by the name "JavacCompilerDaemon" - it suggests, that this class represents the javac deamon - but it doesn't. In my opoininon changing it to JavacDaemonCompiler would help, this name suggests it is a compiler which uses the JavacDaemon.